velox/exec/Operator.cpp

```diff
@@ -294,6 +294,25 @@ OperatorStats Operator::stats(bool clear) {
   return ret;
 }
 
+uint32_t Operator::outputBatchRows() const {
+  const auto& queryConfig = operatorCtx_->task()->queryCtx()->queryConfig();
+
+  auto rowSize = estimatedOutputRowSize();
+  if (rowSize == kUnknownRowSize) {
+    return queryConfig.defaultOutputBatchRows();
+  }
+
+  if (rowSize * 10'000 < queryConfig.preferredOutputBatchBytes()) {
+    return 10'000; // No more than 10'000 rows
+  }
+  return std::max<uint32_t>(
+      queryConfig.preferredOutputBatchBytes() / rowSize, 1);
```

Comment:
Should we check if rowSize is zero or negative?

Reply:
You're right. Missed it.